Items Editing/az_Trim left, right or both item edges via mouse and razor.lua

```diff
@@ -1,29 +1,395 @@
+      
+      --Combo box and check box background
+      reaper.ImGui_PushStyleColor(ctx, reaper.ImGui_Col_FrameBg(), gui_colors.ComboBox.Default)
+      reaper.ImGui_PushStyleColor(ctx, reaper.ImGui_Col_FrameBgHovered(), gui_colors.ComboBox.Hovered)
+      reaper.ImGui_PushStyleColor(ctx, reaper.ImGui_Col_FrameBgActive(), gui_colors.ComboBox.Active)
+      --Combo box drop down list
+      reaper.ImGui_PushStyleColor(ctx, reaper.ImGui_Col_Header(), gui_colors.ComboBox.Default)
+      reaper.ImGui_PushStyleColor(ctx, reaper.ImGui_Col_HeaderHovered(), gui_colors.ComboBox.Hovered)
+      reaper.ImGui_PushStyleColor(ctx, reaper.ImGui_Col_HeaderActive(), gui_colors.ComboBox.Active)
+      
+      local window_flags = reaper.ImGui_WindowFlags_MenuBar()
+      reaper.ImGui_SetNextWindowSize(ctx, W, H, reaper.ImGui_Cond_Once()) -- Set the size of the windows.  Use in the 4th argument reaper.ImGui_Cond_FirstUseEver() to just apply at the first user run, so ImGUI remembers user resize s2
+      
+      reaper.ImGui_PushStyleColor(ctx, reaper.ImGui_Col_Text(), gui_colors.White) 
+      local visible, open = reaper.ImGui_Begin(ctx, 'Options for Trim left or right item edges', true, window_flags)
+      reaper.ImGui_PopStyleColor(ctx, 1)
+      
+      if visible then
+          frame() 
+          if loopcnt == 0 then reaper.ImGui_SetWindowSize(ctx, 0, 0, nil ) end
+          reaper.ImGui_End(ctx)
+      end
+      
+      reaper.ImGui_PopStyleColor(ctx, 13)
+      reaper.ImGui_PopFont(ctx)
+       
+      esc = escMouse or reaper.ImGui_IsKeyReleased(ctx, reaper.ImGui_Key_Escape())
+      enter = enterMouse or reaper.ImGui_IsKeyReleased(ctx, reaper.ImGui_Key_Enter())
+      
+      if ExternalOpen == true then
+        space = spaceMouse or reaper.ImGui_IsKeyReleased(ctx, reaper.ImGui_Key_Space()) 
+        if space == true then SetExtStates() end
+      end
+      
+      if open and esc ~= true and enter ~= true then
+          reaper.defer(loop)
+      elseif enter == true then
+          SetExtStates()
+          reaper.ImGui_DestroyContext(ctx)
+      else
+          reaper.ImGui_DestroyContext(ctx)
+      end
+    loopcnt = loopcnt+1
+  end
+  -----------------
+  local fontName
+  ctx = reaper.ImGui_CreateContext('Options for Trim left or right item edges') -- Add VERSION TODO
+  if reaper.GetOS():match("^Win") == nil then
+    reaper.ImGui_SetConfigVar(ctx, reaper.ImGui_ConfigVar_ViewportsNoDecoration(), 0)
+    fontName = 'sans-serif'
+  else fontName = 'Calibri'
+  end
+  font = reaper.ImGui_CreateFont(fontName, fontSize, reaper.ImGui_FontFlags_None()) -- Create the fonts you need
+  fontSep = reaper.ImGui_CreateFont(fontName, fontSize-2, reaper.ImGui_FontFlags_Italic())
+  if imgui_version_num >= 18910 then
+    reaper.ImGui_Attach(ctx, font)
+    reaper.ImGui_Attach(ctx, fontSep)
+  else
+    reaper.ImGui_AttachFont(ctx, font)
+    reaper.ImGui_AttachFont(ctx, fontSep)
+  end
```

Comment:
This is unreliable. The correct way is enabling ReaImGui's backward-compatibility module before using any of its functions:

```lua
local imgui_path = r.GetResourcePath() .. '/Scripts/ReaTeam Extensions/API/imgui.lua'
if not r.file_exists(imgui_path) then
  -- ...
  return
end

dofile(imgui_path) '0.8.3' -- version check + backward compat

-- from this point onward it's safe to use r.ImGui_* functions (only of v0.8.3 as requested above)
```

Reply:
I don't want to stick to old version. It's just a workaround for the font  function, it works for some range of versions and I guess it's enough, because  most of users keep imgui up to date, thanks to Sexan))
Anyway, thanks for the tip, I'll look how it works.
And hmmm, I have a couple questions.
So, will your way safe for future? If user have installed newer version, which hasn't some of current functions, will script work anyway if i call an older version, or user will get a message to install specified version?

Reply:
> So, will your way safe for future?

Yes, that's the whole point of doing `dofile('imgui.lua')('request_version')` (and why it's important for a public release).

- If the current version < requested version = error
- If the current version > requested version = backward compatibility kicks in to restore the API that the script expects

`reaper.ImGui_GetVersion` before (or without) `dofile('imgui.lua')` is not reliable because it's subject to breaking changes just like the rest of the API (eg. v0.7 changed the order of return values.

Reply:
![Screen Shot 2024-03-12 at 03 51 52](https://github.com/ReaTeam/ReaScripts/assets/4297676/737f0e11-d9e0-464b-9781-b79544d67fa8)

Reply:
> eg. v0.7 changed the order of its return values

Hmm, but my script works for version 0.7 as well.

Do I understand right that dofile('imgui.lua')  sticks  the script to a particular version and I need to choose main version for my script rather than add behavior for another versions?